src/main/java/core/basesyntax/dao/impl/MessageDaoImpl.java

```diff
@@ -2,31 +2,79 @@
 
 import core.basesyntax.dao.MessageDao;
 import core.basesyntax.model.Message;
+import java.util.ArrayList;
 import java.util.List;
+import javax.persistence.Query;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
 
 public class MessageDaoImpl extends AbstractDao implements MessageDao {
     public MessageDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
-    public Message create(Message entity) {
-        return null;
+    public Message create(Message message) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(message);
+            transaction.commit();
+            return message;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't create a message: " + message, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Message get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Message.class, id);
+        } catch (Exception e) {
+            throw new RuntimeException("Couldn't get message by id: " + id, e);
+        }
     }
 
     @Override
     public List<Message> getAll() {
-        return null;
+        List<Message> messages = new ArrayList<>();
+        try (Session session = factory.openSession()) {
+            Query query = session.createQuery("FROM Message", Message.class);
+            messages.addAll(query.getResultList());
+        } catch (Exception e) {
+            throw new RuntimeException("Couldn't get all messages", e);
+        }
+        return messages;
```

Comment:
you can make here one return only 
```suggestion
        try (Session session = factory.openSession()) {
            return session.createQuery("FROM Message ", Message.class).getResultList();
        } catch (Exception e) {
            throw new RuntimeException("Couldn't get all messages", e);
        }
```